pkg/array/array.go

```diff
@@ -412,3 +537,71 @@ func GetVolumeUUIDPrefix(volumeID string) (prefix string) {
 
 	return prefix
 }
+
+// GetLeastUsedActiveNAS finds the active NAS with the least FS count
+func GetLeastUsedActiveNAS(ctx context.Context, arr *PowerStoreArray, nasServers []string) (string, error) {
```

Comment:
I see in single method many checks, is it possible to break down large functions into smaller for effective usage?